### PR TITLE
Reorganize preference components

### PR DESCRIPTION
**Pull request by SuperDragonXD**
## Description

This PR aims to reorganize the preference components to make them more clear on what each file does. I have also added a README file to better explain each folder.

See also the [similar PR](https://github.com/LawnchairLauncher/lawnicons/pull/926) at Lawnicons (but this PR only contains moved files).

## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lawnchair/src/app/lawnchair/ui/preferences/components/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# About the `components` directory
+The `components` directory should be organized into the following:
+
+* For components for a specific UI that require many files, add it in a subdirectory. Example: `colorpreference`
+* For components that allow the user to control/select something (a list, yes/no, etc), add it in the `controls` subdirectory.
+* For components that are related to the layout itself and unrelated to the above, add it in the `layout` directory.
+* Anything else, you can add it in the current `./` directory.
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/ui/preferences/components/README.md:7]: Should we leave this md file under sourceSet?

**Review by Goooler — APPROVED** [review]

**Review by MrSluffy — APPROVED** [review]

File: lawnchair/src/app/lawnchair/ui/preferences/components/README.md (hunk 1 above)
Comment (SuperDragonXD) [reply lawnchair/src/app/lawnchair/ui/preferences/components/README.md:7]: Should be fine. (Not sure actually)

Comment (Goooler) [reply lawnchair/src/app/lawnchair/ui/preferences/components/README.md:7]: May not add this.

**Comment by MrSluffy** [comment]
@SuperDragonXD don't merge this for now as I already merge and working in A14 Thanks

**Review by suphon-t — APPROVED** [review]

**Pull request merged**